### PR TITLE
Add luacheck

### PR DESCRIPTION
**Pull request by olegrok**
* Add luacheck
* Add simple Makefile
* Fix warnings: line is too long
* Fix:
```lua
tarantool> icu.parse({cal = box.NULL}, '')
Segmentation fault
  code: SEGV_MAPERR
  addr: 0x74
  context: 0x10a41fac8
  siginfo: 0x10a41fa60
Current time: 1551415142
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.PHONY: lint
+lint:
+	luacheck --config=.luacheckrc --no-unused-args --no-redefined \
+	                    ./icu-date.lua ./test
```
Comment (Totktonada) [inline Makefile]: Maybe add other lua files too?

File: icu-date.lua [hunk 2]
```diff
@@ -63,7 +63,7 @@ end
 
 local function check_self_type(self, func)
   assert(type(self) == "table", ("Use date:%s(...) instead of date.%s(...)"):format(func, func))
-  assert(self.cal, "Calendar is not specified")
+  assert(self.cal ~= nil, "Calendar is not specified")
```
Comment (Totktonada) [inline icu-date.lua]: I think it should be factored out to a separate commit.

File: Makefile (hunk 1 above)
Comment (olegrok) [reply Makefile]: Done

The exception to this is the `spec` directory, because it seems to use luaunit that overrides global variables

File: icu-date.lua (hunk 2 above)
Comment (olegrok) [reply icu-date.lua]: Done
